frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx

```diff
@@ -0,0 +1,166 @@
+    pipelineId,
+    initialData,
+    initialPageToken,
+    sortDirection,
+    sortField,
+    filter,
+    loaded,
+  });
+
+  const { totalSize, ...searchProps } = useSelectorSearch({ setFilter, fetchedSize, loaded });
+  const { onClear } = searchProps;
+
+  const menu = (
+    <Menu data-id="pipeline-version-selector-menu" ref={menuRef} isScrollable>
+      <MenuContent>
+        <MenuSearch>
+          <MenuSearchInput>
+            <SearchInput {...searchProps} aria-label="Filter pipeline versions" />
+          </MenuSearchInput>
+          <HelperText>
+            <HelperTextItem variant="indeterminate">{`Type a name to search your ${totalSize} versions.`}</HelperTextItem>
+          </HelperText>
+        </MenuSearch>
+        <MenuList>
+          <div role="menuitem">
+            <TableBase
+              {...tableProps}
+              itemCount={fetchedSize}
+              loading={!loaded}
+              data-id="pipeline-version-selector-table-list"
+              emptyTableView={<EmptySelectorView onClearFilters={onClear} />}
+              borders={false}
+              variant={TableVariant.compact}
+              columns={pipelineVersionSelectorColumns}
+              data={versions}
+              rowRenderer={(row, index) => (
+                <PipelineSelectorTableRow
+                  key={index}
+                  obj={row}
+                  onClick={() => {
+                    onSelect(row);
+                    setOpen(false);
+                  }}
+                />
+              )}
+              getColumnSort={getSelectorTableColumnSort({
+                columns: pipelineVersionSelectorColumns,
+                ...tableProps,
+              })}
+              footerRow={() =>
+                loaded ? (
+                  <PipelineViewMoreFooterRow
+                    visibleLength={versions.length}
+                    totalSize={fetchedSize}
+                    errorTitle="Error loading more pipeline versions"
+                    onClick={onLoadMore}
+                    colSpan={2}
+                  />
+                ) : null
+              }
+            />
+          </div>
+        </MenuList>
+      </MenuContent>
+    </Menu>
+  );
+
+  return (
+    <MenuContainer
+      isOpen={isOpen}
+      toggleRef={toggleRef}
+      toggle={
+        <MenuToggle
+          id="pipeline-version-selector"
+          icon={
+            pipelineId &&
+            !initialLoaded && (
+              <Icon>
+                <Spinner size="sm" aria-label="Loading pipeline versions" />
+              </Icon>
+            )
+          }
+          style={{ maxWidth, minWidth }}
+          ref={toggleRef}
+          onClick={() => setOpen(!isOpen)}
+          isExpanded={isOpen}
+          isDisabled={!pipelineId || totalSize === 0}
+          isFullWidth
+          data-testid="pipeline-version-toggle-button"
+        >
+          {!pipelineId
```

Comment:
Yes, this is an agreement by me and @christianvogt. When there is no pipeline id, all we want to remind the user is to choose a pipeline first (on Pipeline runs page)

Reply:
Right, but this component is independent of the pipeline runs page and is used in different places where that might not make sense (anywehere that doesn't have a pipeline input right next to this selector).

Reply:
I'm not sure if this makes sense to check for a pipeline ID and show a message at all for that condition, but if we absolutely want to check for a pipelineId, I assume if 1 is not found then "no versions available" would make more sense, since there would therefore be no versions returned from the API response?

Reply:
I can change that, though it's on the mockup saying `Select a pipeline version`

<img width="1185" alt="Screenshot 2023-12-15 at 12 44 22 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/632b3e62-bcf0-47d9-b910-9accc7787aec">

Reply:
I think the placeholder logic should be to show "Select a pipeline version" by default, and on this page, if no pipeline is selected (and therefore no pipelineId passed to the selector for the version), the input would be disabled.